executor/oci/resolvconf.go

```diff
@@ -51,6 +51,22 @@ func GetResolvConf(ctx context.Context, stateDir string, idmap *idtools.Identity
 				} else {
 					if fi.ModTime().Before(fiMain.ModTime()) {
 						generate = true
+					} else {
+						var oriFileHasLocalDNS bool
+						if ori, err := resolvconfGet(); err == nil {
+							oriFileHasLocalDNS = resolvconf.HasLocal(ori.Content[:])
+						}
+
+						var genFileHasLocalDNS bool
+						if gen, err := resolvconf.GetSpecific(p); err == nil {
+							genFileHasLocalDNS = resolvconf.HasLocal(gen.Content[:])
+						}
+
+						if hostNetMode && oriFileHasLocalDNS && !genFileHasLocalDNS {
```

Comment:
After comparing all possible combinations, only these 2 combinations need to re-generate `resolv.conf`.